### PR TITLE
keycloak-updates

### PR DESCRIPTION
**Pull request by philbrookes**
- Update keycloak-operator.
- Default to createOnly true in openshift keycloakrealm
- create upgrade playbook for 1.3 > 1.4

Jiras:
INTLY-1680
INTLY-1724
INTLY-1679

### CODE REVIEW

**Comment by maleck13** [comment]
**Verification**

1) install a 1.3 cluster
2) run the upgrade playbook verify that customer-admin has the correct roles and can still login to sso 
3) add a new role to the user and a new idp ensure they are not removed by the operator
4) uninstall and install from scratch ensure the same as above but on a freshly installed cluster

**Comment by maleck13** [comment]
👁

File: playbooks/upgrades/1.3_to_1.4.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: Update keycloak-operator image tag
+      shell: "oc patch deployment keycloak-operator -n sso --type json -p '[{\"op\": \"replace\", \"path\": \"/spec/template/spec/containers/0/image\", \"value\": \"quay.io/integreatly/keycloak-operator:v1.3.4\"}]'"
```
Comment (maleck13) [inline playbooks/upgrades/1.3_to_1.4.yaml]: The namespace can't be hard coded here

File: playbooks/upgrades/1.3_to_1.4.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+
+    - name: add CreateOnly to openshift keycloak realm
+      shell: "oc patch keycloakrealm openshift -n sso --type json -p '[{\"op\": \"add\", \"path\": \"/spec/createOnly\", \"value\": true}]'"
```
Comment (maleck13) [inline playbooks/upgrades/1.3_to_1.4.yaml]: same here

**Comment by philbrookes** [comment]
SOP: https://github.com/integr8ly/help/pull/108

**Comment by philbrookes** [comment]
cannot re-open this for some reason, its here: https://github.com/integr8ly/help/pull/108

**Pull request closed without merging**